### ISSUE TITLE
Istanbul

### ISSUE THREAD

**Issue by eranshapira** [zxbodya/angular-webpack-seed#3]
Hey, Great job with the seed.
What about adding Coverage for tests? I'm working on it atm, was wondering if you had any progress.

**Comment by zxbodya** [comment]
Hi,
Have tried to add coverage on very similar karma configuration, there is latest version of that:

https://github.com/zxbodya/di1/tree/64b934d6be578f0d4826c0294f1aa3e9a304b1cc

Generally it works, but it has some issues:

1. report is about transpiled code(not es6 sources)
2. report does not include files that was not tested at all

Pushed it to branch https://github.com/zxbodya/angular-webpack-seed/tree/feature/coverage-from-d1

Since that particular project does not require running in browser - I stopped using karma test runner, used Istanbul with jasmine directly - that solved second issue for me.

Later, I solved first issue with `remap-istanbul`.

---

There is also project `isparta` - it looks more suitable for this case.

But maybe it will require switch to babel6 (I have done it for project with almost the same build configuration - later today, will try to apply that updates here)

**Comment by zxbodya** [comment]
Updated babel dependency to version 6.
Added code coverage report using [babel-istanbul-loader](https://github.com/deepsweet/babel-istanbul-loader) - works really well :)

Also improved Travis CI configuration, and added coveralls and codecov.io badges.

---

Also included all source files in branch `feature/coverage-from-d1`. Planning to remove it soon - let me know when you do not need it.

**Comment by lxeran** [comment]
omg, i can't believe you already did it, i was just about to tell you that I managed to pull it off :+1: 
:smile: well, i'll take a look at what you did! i already did the same thing (upgraded to babel6 and used isparta) - maybe i'll merge your changes instead, if not - i'll open a PR for you to look.
great job man, really nice.

**Comment by zxbodya** [comment]
`babel-istanbul-loader ` can be literally replaced with `isparta-loader`(pushed it to `feature/isparta` branch ) - both are from the same author, and eventually have exactly the same interface.

Tried both, and did not noticed any difference. Initially I started with `babel-istanbul-loader` because of note about it in `isparta-loader` readme file. Also `babel-istanbul` claims to `run exactly like istanbul` - so it looks more straightforward solution.

If you will find any benefits for switching to isparta over babel-istanbul - let me know.
